datafn/python/datafn/handlers/mutation.py

```
@@ -0,0 +1,163 @@
+        match = await evaluate_guard(db, resource, record_id, guard)
+        if not match:
+            raise DatafnError(code="CONFLICT", message="Guard condition not met", details={"path": "if"})
+    
+    # Execution
+    record = payload.get("record", {})
+    affected_ids = []
+    
+    if operation == "insert":
+        # Validate fields
+        err = validate_record_keys(index, resource, record, "record")
+        if err: raise err
```

Ruff (0.14.13)

Multiple statements on one line (colon)

(E701)